glue/core/component.py

```diff
@@ -338,6 +338,10 @@ def __setgluestate__(cls, rec, context):
         return cls(None, rec['axis'], rec['world'])
 
 
+class AutoDerivedComponent(DerivedComponent):
```

Comment:
small nit: I'm not sure if you want the name of this to be tied to its origin (added by machine) or its functional organization (should be invisible in certain contexts)